site/source/tutorials/10XX_development_environment/index.md

```diff
@@ -0,0 +1,146 @@
+
+Within the demo app, open `src/widgets/WorkerContainer.ts`. In it, there should be code to create an instance of the Worker widget on line 21. Add a fake property, e.g.:
+
+```typescript
+const workers = workerData.map((worker, i) => w(Worker, {
+  key: `worker-${i}`,
+  fakeProp: 'foo',
+  ...worker
+}));
+```
+
+Verify that your editor has highlighted `fakeProp` as an error:
+
+<p class="center">![error message in Atom](./resources/atom_ts_error.png)</p>
+
+{% instruction 'Go further: VS Code and tasks' %}
+
+You may skip to the next step if you are not interested in trying out VS Code.
+
+VS Code supports Typescript by default, is written is Typescript, and allows you to [configure tasks](https://code.visualstudio.com/docs/editor/tasks) which are then integrated into the IDE.
+
+To configure tasks:
+
+* Press <kbd>⌘⇧P</kbd>/<kbd>Ctrl⇧P</kbd> to bring up the command list.
+* Start typing <kbd>tasks</kbd> and select the command `Tasks: Configure Task Runner`
+
+<p class="center">![gif of opening tasks.json](./resources/task_setup.gif)</p>
+
+The [first tutorial](../000_local_installation/) had you run `dojo build` as the first Dojo CLI command after creating your app, so consider the following code to configure it within `tasks.json`:
+
+```json
+{
+  "version": "0.1.0",
+  "command": "dojo",
+  "isShellCommand": true,
+  "args": [],
+  "showOutput": "always",
+  "suppressTaskName": true,
+  "tasks": [
+    {
+      "taskName": "build",
+      "args": [ "build" ],
+      "isBuildCommand": true,
+      "problemMatcher": {
+      "owner": "dojo",
+      "fileLocation": "relative",
+      "pattern": [
+        {
+          "regexp": "^(\\S+) in (.*)",
+          "severity": 1,
+          "file": 2
+        }, {
+          "regexp": "\\((\\d+),(\\d+)\\):(.*)",
+          "line": 1,
+          "column": 2,
+          "message": 3
+        }
+      ]
+      }
+    }
+  ]
+}
+```
+
+Other tasks to consider adding could be `test` or `watch`. Once configured, the build command (<kbd>⌘⇧B</kbd>/<kbd>Ctrl⇧B</kbd>) will be configured as well as you will have the other
+tasks available in the IDE:
+
+<p class="center">![dropdown of vs code tasks](./resources/run_task.png)</p>
+
+{% section %}
+
+## Debugging
+
+If you are developing a Dojo 2 application in TypeScript, your application code is transpiled from TypeScript to JavaScript. If you are also using dojo build, then your code is also being bundled and minimized. This could make it challenging to debug an application, but we have tried to make Dojo 2 applications integrate well into modern debugging tools.
+
+Source maps are the way on the web we can describe code that has been transformed in some fashion. The dojo build maps the code throughout the process so that both the original TypeScript code and CSS code is available when debugging. This should allow you to set breakpoints and watches on the original code as well as see the original code when there is a run-time error.
```

Comment:
Source maps are the way on the web we can describe code that has been transformed in some fashion. -> Source maps describe a way to map from transformed source code back to its original source.